addons/mail/static/src/new/public/discuss_public.js

```diff
@@ -45,4 +46,13 @@ export class DiscussPublic extends Component {
             serverData: { uuid: this.props.data.channelData.uuid },
         });
     }
+
+    get guest() {
+        if (this.store.currentGuest?.id) {
+            return Guest.insert(this.store, {
+                id: this.store.currentGuest?.id,
+            });
+        }
+        return undefined;
+    }
```

Comment:
Not needed at all, can rely on `store.currentGuest` completely

Reply:
same as previously, maybe I can make this check when creating currentGuest